src/_igraph/attributes.c

```diff
@@ -1491,7 +1493,7 @@ static int igraphmodule_i_attribute_get_info(const igraph_t *graph,
     if (n) {
       retval = igraphmodule_PyList_to_strvector_t(keys, n);
       if (retval) {
-        return retval;
+        return (igraph_error_t)retval;
```

Comment:
Casting is not correct here; `igraphmodule_PyList_to_strvector_t` returns 0 for success and 1 for error (similarly to many functions in Python's C API). You should do something like `retval ? IGRAPH_FAILURE : IGRAPH_SUCCESS`, which communicates the intention more clearly.

Reply:
yep, good point